pkg/controller/common.go

```diff
@@ -741,30 +624,44 @@ func (s *scoreByNodeCapacity) name() string {
 	return "score by node capacity"
 }
 
-func (s *scoreByNodeCapacity) setup(pods []v1.Pod, volumes []string, state *SharedState) {
+func (s *scoreByNodeCapacity) priority() float64 {
+	return 1
+}
+
+func (s *scoreByNodeCapacity) setup(nodes []v1.Node, pods []v1.Pod, volumes []string, state *SharedState) {
+	s.nodes = nodes
```

Comment:
It seems like we're only using the node name. Could we simply store the names instead of a full copy of the pods to reduce the memory usage?

Reply:
Pod objects are used in other score plugins as `scoreByInterPodAffinity`
https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/6120a76ee821b08db119deab10b8d155d4f415ff/pkg/controller/common.go#L759-L778
And node objects are used in `scoreByPodNodeAffinity`
https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/6120a76ee821b08db119deab10b8d155d4f415ff/pkg/controller/common.go#L888-L893